File: favourite-lyrics-core/src/main/java/org/jesperancinha/lyrics/core/service/LyricsServiceImpl.java
package org.jesperancinha.lyrics.core.service;

import org.jesperancinha.lyrics.domain.data.LyricsDto;
import org.jesperancinha.lyrics.domain.data.LyricsFullDto;
import org.jesperancinha.lyrics.domain.port.LyricsPersistencePort;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.UUID;

@Service
public class LyricsServiceImpl implements LyricsService {

    private final LyricsPersistencePort lyricsPersistencePort;

    public LyricsServiceImpl(LyricsPersistencePort lyricsPersistencePort) {
        this.lyricsPersistencePort = lyricsPersistencePort;
    }

    @Override
    public void addLyrics(LyricsDto lyricsDto) {
        lyricsPersistencePort.addLyrics(lyricsDto);
    }

    @Override
    @Transactional
    public void removeLyrics(LyricsDto lyricsDto) {
        lyricsPersistencePort.removeLyrics(lyricsDto);
    }

    @Override
    public void updateLyrics(LyricsDto lyricsDto) {
        lyricsPersistencePort.updateLyrics(lyricsDto);
    }

    @Override
    public List<LyricsDto> getAllLyrics() {
        return lyricsPersistencePort.getAllLyrics();
    }

    @Override
    public List<LyricsFullDto> getAllFullLyrics() {
        return lyricsPersistencePort.getAllLFullLyrics();
    }

    @Override
    public LyricsDto getLyricsById(UUID lyricsId) {
        return lyricsPersistencePort.getLyricsById(lyricsId);
    }
}

File: favourite-lyrics-core/src/main/kotlin/org/jesperancinha/lyrics/core/service/LyricsServiceImpl.kt
package org.jesperancinha.lyrics.core.service

import org.jesperancinha.lyrics.domain.data.LyricsDto
import org.jesperancinha.lyrics.domain.data.LyricsFullDto
import org.jesperancinha.lyrics.domain.port.LyricsPersistencePort
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import java.util.*

@Service
class LyricsServiceImpl(private val lyricsPersistencePort: LyricsPersistencePort) : LyricsService {
    override fun addLyrics(lyricsDto: LyricsDto?) {
        lyricsPersistencePort.addLyrics(lyricsDto)
    }

    @Transactional
    override fun removeLyrics(lyricsDto: LyricsDto?) {
        lyricsPersistencePort.removeLyrics(lyricsDto)
    }

    override fun updateLyrics(lyricsDto: LyricsDto?) {
        lyricsPersistencePort.updateLyrics(lyricsDto)
    }

    override val allLyrics: List<LyricsDto?>?
        get() = lyricsPersistencePort.allLyrics
    override val allFullLyrics: List<LyricsFullDto?>?
        get() = lyricsPersistencePort.allLFullLyrics

    override fun getLyricsById(lyricsId: UUID?): LyricsDto? {
        return lyricsPersistencePort.getLyricsById(lyricsId)
    }
}